### ISSUE TITLE
tidb-monitor grafana should set username and password in a Secret

### ISSUE THREAD

**Issue by yeya24** [pingcap/tidb-operator#2637]
## Feature Request

**Is your feature request related to a problem? Please describe:**

From the example https://github.com/pingcap/tidb-operator/blob/master/manifests/monitor/tidb-monitor.yaml, seems the grafana in tidb-monitor can only set username and password in plain text. This is not good and it is better to use `Secret` to specify these fields.

**Comment by tennix** [comment]
Good point! For secret data, it definitely needs to be stored in Secret instead of plaintext in pod spec.

**Comment by DanielZhangQD** [comment]
@yeya24 would you like to submit PR for this issue?

**Comment by yeya24** [comment]
> @yeya24 would you like to submit PR for this issue?

I am not sure about the time. But I can work on it if no one else wants to take it.

**Comment by DanielZhangQD** [comment]
@yeya24 Great! Thanks!

**Comment by DanielZhangQD** [comment]
Partial work has been done in #2656.
